### PR TITLE
Fixes #5416 Reduced UsedCss index key size

### LINKED ISSUE #5416: Limit the index column size for queue_name
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version
- Used the search feature to ensure that the bug hasn’t been reported before

**Describe the bug**
For specific table encoding we might encounter the following error:
```
[08-Sep-2022 18:02:15 UTC] WordPress database error Index column size too large. The maximum column size is 767 bytes. for query CREATE TABLE wp_wpr_rucss_used_css ( 
            id               bigint(20) unsigned NOT NULL AUTO_INCREMENT,
            url              varchar(2000)       NOT NULL default '',
            css              longtext                     default NULL,
            hash             varchar(32)                  default '',
            unprocessedcss   longtext                NULL,
            retries          tinyint(1)          NOT NULL default 1,
            is_mobile        tinyint(1)          NOT NULL default 0,
            job_id           varchar(255)        NOT NULL default '',
            queue_name       varchar(255)        NOT NULL default '',
            status           varchar(255)        NOT NULL default '',
            modified         timestamp           NOT NULL default '0000-00-00 00:00:00',
            last_accessed    timestamp           NOT NULL default '0000-00-00 00:00:00',
            PRIMARY KEY (id),
            KEY url (url(150), is_mobile),
            KEY modified (modified),
            KEY last_accessed (last_accessed),
            INDEX `queue_name_index` (`queue_name`),
            KEY hash (hash) ) DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_520_ci made by require_once('wp-admin/admin.php'), do_action('admin_init'), WP_Hook->do_action, WP_Hook->apply_filters, WP_Rocket\Dependencies\Database\Table->maybe_upgrade, WP_Rocket\Dependencies\Database\Table->install, WP_Rocket\Dependencies\Database\Table->create
```

**Expected behavior**
We should limit `INDEX 'queue_name_index' ('queue_name'),` to be maximum 191 characters. 

**Additional context**
We need to decide what's maximum expected size there and alter tables for existing installations.

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

### PR DESCRIPTION
**Pull request by CrochetFeve0251**
## Description

Fix a bug at the creation from the RUCSS table due to the index key being too long.
For that we reduced to 191.

Fixes #5416

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

## Is the solution different from the one proposed during the grooming?

No.

## How Has This Been Tested?

- [ ] Manually

# Checklist:

Please delete the options that are not relevant.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by remyperona — COMMENTED** [review]
don't we also need to update the schema for existing installs?

**Comment by piotrbak** [comment]
@Tabrisrp What do you think about reducing the size here:
https://github.com/wp-media/wp-rocket/blob/c6b74a8ad463158511a7d10352554379d4731099/inc/Engine/Optimization/RUCSS/Database/Tables/UsedCSS.php#L68

That's what @engahmeds3ed suggested at some point, not sure if that works and which option is better/cleaner.

**Comment by remyperona** [comment]
It really depends on the possible length of the queue_name values, do we know how long it can be?

**Comment by piotrbak** [comment]
From what I can see, we should not expect this value to be long:
image

**Comment by remyperona** [comment]
Makes me wonder why the queue name is indexed in the first place, we don't query based on it in the plugin. Looking at the code, I think there was a mistake at some point, making this an index instead of the status column.

**Comment by CrochetFeve0251** [comment]
> Makes me wonder why the queue name is indexed in the first place, we don't query based on it in the plugin. Looking at the code, I think there was a mistake at some point, making this an index instead of the status column.

Ok I think for that I will have to update the schema for all customers but I agree it is weird to index on that value.

**Comment by piotrbak** [comment]
Since it's scheduled for 3.12.2 we have some time for discussion. 
```
making this an index instead of the status column.
```
Which queries would take advantage of this change? 🤔

**Comment by remyperona** [comment]
We have some queries that are based on the current status, so if there is a lot of entries, they could be faster

**Review by remyperona — CHANGES_REQUESTED** [review]
I think we need to add the method to update the existing schema on update

**Review by remyperona — CHANGES_REQUESTED** [review]
We should also drop the index for ` the queue_name` in the update method

File: inc/Dependencies/RocketLazyload/Iframe.php [hunk 1]
```diff
@@ -106,7 +106,6 @@ private function getExcludedPatterns() {
 				'loading="eager"',
 				'data-skip-lazy',
 				'skip-lazy',
-				'google_ads_iframe_',
```
Comment (remyperona) [inline inc/Dependencies/RocketLazyload/Iframe.php]: This shouldn't be modified

**Review by remyperona — CHANGES_REQUESTED** [review]

**Review by remyperona — APPROVED** [review]

**Review by remyperona — CHANGES_REQUESTED** [review]
getting the following error locally:  `You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'DELETE INDEX `queue_name_index` (`status`)' at line 1`

**Review by remyperona — APPROVED** [review]

**Comment by vmanthos** [comment]
@CrochetFeve0251 When installing WP Rocket from scratch indexes look fine.

However, updating WP Rocket from a previous version changes the `queue` index to `status` but it doesn't have the `191` length restriction.

I believe this is coming from here:

https://github.com/wp-media/wp-rocket/blob/35a401ed45acb41d42e19943f1030d3b5cb618ae/inc/Engine/Optimization/RUCSS/Database/Tables/UsedCSS.php#L216

Can you please look into this? 🙏

**Review by vmanthos — APPROVED** [review]
Working as expected!
[testrail-report-416.pdf](https://github.com/wp-media/wp-rocket/files/9754572/testrail-report-416.pdf)

**Pull request merged**